apps/wappush/js/messagedb.js

```diff
@@ -76,18 +102,27 @@ var MessageDB = (function() {
 
       req.onsuccess = function mdb_opened(event) {
         db = event.target.result;
-        resolve(db);
+        mdb_deleteViewedMessages().then(() => resolve(db));
       };
       req.onerror = function mdb_openError(event) {
         reject(event.target.error);
       };
       req.onupgradeneeded = function mdb_upgradeNeeded(event) {
         var db = event.target.result;
-
-        if (event.oldVersion === 0) {
-          var store = db.createObjectStore(MESSAGE_STORE_NAME,
-                                           { keyPath: KEY_PATH });
-          store.createIndex(ID_INDEX, ID_INDEX_KEYPATH, { unique: false });
+        var store = null;
+
+        switch (event.oldVersion) {
+          case 0:
+            store = db.createObjectStore(MESSAGE_STORE_NAME,
+                                         { keyPath: KEY_PATH });
```

Comment:
nit: I can see you spent some time in Gecko looking at your indentation style.

Here is what I'm used to, tell me what you think:
```javascript
store = db.createObjectStore(
  MESSAGE_STORE_NAME, { keyPath: KEY_PATH }
);
```
The idea is that I try to never increase indentation by more than 2 spaces.
Especially I don't want to have indentation depend on variable/function names, which is the case here.